File: Week_01/917.reverse-only-letters/Solution.java
class Solution {
    public String reverseOnlyLetters(String S) {
        char[] arr = S.toCharArray();
        int l = 0, r = arr.length - 1;
        while (l < r) {
            boolean isLetterL = Character.isLetter(arr[l]);
            boolean isLetterR = Character.isLetter(arr[r]);
            if (isLetterL && isLetterR) {
                char temp = arr[l];
                arr[l] = arr[r];
                arr[r] = temp;
                l++;
                r--;                
                continue;
            }
            if (!isLetterL) {
                l++;
            }
            if (!isLetterR) {
                r--;
            }
        }
        return String.valueOf(arr);
    }
}

File: Week_01/917.reverse-only-letters/Solution.py
class Solution:
    def reverseOnlyLetters(self, S: str) -> str:
        arr = list(S)
        l, r = 0, len(S) - 1
        while l < r:
            lIsAlphabet = arr[l].isalpha()
            rIsAlphabet = arr[r].isalpha()

            if lIsAlphabet and rIsAlphabet:
                arr[l], arr[r] = arr[r], arr[l]
                l +=1
                r -=1
                continue;

            if not lIsAlphabet:
                l += 1
            if not rIsAlphabet:
                r -= 1

        return "".join(arr)
    